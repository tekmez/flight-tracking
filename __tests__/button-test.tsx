import TButton from "@/components/ui/Button";
import * as React from "react";
import { fireEvent, render } from "@testing-library/react-native";

describe("Button", () => {
  it("renders correctly", () => {
    const tree = render(
      <TButton
        title="Go to Flights"
        onPress={() => {}}
        classNames="bg-orange-600 mt-4"
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("onPress works correctly", () => {
    const mockFn = jest.fn();
    const button = render(
      <TButton
        title="Go to Flights"
        onPress={mockFn}
        classNames="bg-orange-600 mt-4"
      />
    );
    fireEvent.press(button.getByText("Go to Flights"));
    expect(mockFn).toHaveBeenCalled();
  });
});
