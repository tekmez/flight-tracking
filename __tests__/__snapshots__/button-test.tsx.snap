// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "borderBottomLeftRadius": 14,
      "borderBottomRightRadius": 14,
      "borderTopLeftRadius": 14,
      "borderTopRightRadius": 14,
      "opacity": 1,
      "paddingBottom": 14,
      "paddingLeft": 20,
      "paddingRight": 20,
      "paddingTop": 14,
    }
  }
>
  <Text
    style={
      [
        {
          "fontFamily": "SFProRegular",
        },
        {
          "color": "#fff",
        },
        {
          "fontSize": 17,
        },
        {
          "lineHeight": 22,
        },
      ]
    }
  >
    Go to Flights
  </Text>
</View>
`;
